jerry-core/jerry.c

```diff
@@ -1698,6 +1698,8 @@ jerry_parse (const jerry_api_char_t *source_p, /**< script source */
 {
   jerry_assert_api_available ();
 
+  vm_finalize ();
```

Comment:
I tried to trace down similarities/differences between global and eval run, and it turned out that they cannot be forced to act the same. The standard (ECMA-262 5.1ed) has its relevant parts in section 10.4. Even "if the eval code is not being evaluated by a direct call" and we "initialise the execution context as if it was a global execution context", when performing "Declaration Binding Instantiation", we come to a difference that cannot be worked around: "If code is eval code, then let configurableBindings be true else let configurableBindings be false". Our implementation follows these instructions (fortunately).

Long story cut short, eval'd code will always be executed somewhat differently from globally executed code. Thus, it's no good for us.